clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -691,6 +691,33 @@ check_rseq_side_exit_discontinuity()
     return true;
 }
 
+bool
+check_timestamp_increase_monotonically(void)
+{
+    // Positive test: timestamps increase monotonically.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_marker(1, TRACE_MARKER_TYPE_TIMESTAMP, 0),
+            gen_marker(1, TRACE_MARKER_TYPE_TIMESTAMP, 10),
+            gen_marker(1, TRACE_MARKER_TYPE_TIMESTAMP, 10),
```

Comment:
Hmm, I believe "monotonically" implies values do not stay the same.  Are you worried about two identical real timestamps in a row?  I don't think this will happen with our microsecond timestamps on the current platforms; it seems it would only happen on some platform without a reliable OS-provided time interface but there we would probably use the processor timestamp counter.

Reply:
Based on 

https://en.wikipedia.org/wiki/Monotonic_function: 
A function is called monotonically increasing (also increasing or non-decreasing)[3] if for all x and y such that x <=y one has f(x) <= f(y), so f preserves the order.

https://en.wiktionary.org/wiki/monotonic_increasing#:~:text=monotonic%20increasing%20(not%20comparable),contrast%20this%20with%20strictly%20increasing:
(mathematics, of a function) always increasing or remaining constant, and never decreasing; contrast this with strictly increasing

And Wolfram MathWorld has the definition you mention:
https://mathworld.wolfram.com/MonotoneIncreasing.html
Always increasing; never remaining constant or decreasing. Also called strictly increasing.

Anyway, let me update the code to ensure the timestamps are increasing, and do not stay the same.

Reply:
I've updated the code with a second commit, https://github.com/DynamoRIO/dynamorio/pull/6037/commits/1f90288d7775034fad0e96add549c2f4685f965b.

I forgot to put a comment here. :o(

Reply:
> I forgot to put a comment here. :o(

I think you mean, you forgot to re-request review: there is a circular arrow in the upper right of this page for that.  I just clicked on it.